.devcontainer/devcontainer.json

```diff
@@ -1,9 +1,11 @@
 {
   "name": "Kubebuilder DevContainer",
-  "image": "golang:1.22",
+  "image": "mcr.microsoft.com/devcontainers/go:1.23-bookworm",
```

Comment:
why?

Reply:
Thanks for the question, as mentioned in the commit message this aligns the devcontainer used across `etcd-io/etcd` and `etcd-io/etcd-operator`.  I don't have particularly strong opinions on what the devcontainer actually is so longs it is debian based and has sensible tooling included. Where it makes sense we should align it across the projects.